libbeat/beat/beat.go

```diff
@@ -195,6 +216,9 @@ func (b *Beat) Run() {
 }
 
 // Stop calls the beater Stop action.
+// It can happen that this function is called more then once.
 func (beat *Beat) Stop() {
+	logp.Info("Stopping Beat")
 	beat.BT.Stop()
+	close(beat.done)
```

Comment:
This method is also called after Start() -> see comment above. I think important here is that it is possible to "stop" the beater without stopping the beat itself. Not sure if that would also be covered by the above waitgroup suggestion.